pgcli/packages/sqlcompletion.py

```diff
@@ -53,46 +101,40 @@ def suggest_type(full_text, text_before_cursor):
     if full_text.startswith('\\i '):
         return (Path(),)
 
-    word_before_cursor = last_word(text_before_cursor,
-            include='many_punctuations')
-
-    identifier = None
-
-    def strip_named_query(txt):
-        """
-        This will strip "save named query" command in the beginning of the line:
-        '\ns zzz SELECT * FROM abc'   -> 'SELECT * FROM abc'
-        '  \ns zzz SELECT * FROM abc' -> 'SELECT * FROM abc'
-        """
-        pattern = re.compile(r'^\s*\\ns\s+[A-z0-9\-_]+\s+')
-        if pattern.match(txt):
-            txt = pattern.sub('', txt)
-        return txt
-
-    full_text = strip_named_query(full_text)
-    text_before_cursor = strip_named_query(text_before_cursor)
-    text_before_cursor_including_last_word = text_before_cursor
-
-    # If we've partially typed a word then word_before_cursor won't be an empty
-    # string. In that case we want to remove the partially typed string before
-    # sending it to the sqlparser. Otherwise the last token will always be the
-    # partially typed string which renders the smart completion useless because
-    # it will always return the list of keywords as completion.
-    if word_before_cursor:
-        if word_before_cursor[-1] == '(' or word_before_cursor[0] == '\\':
-            parsed = sqlparse.parse(text_before_cursor)
-        else:
-            text_before_cursor = text_before_cursor[:-len(word_before_cursor)]
-            parsed = sqlparse.parse(text_before_cursor)
+    doc = SqlDocument(full_text, text_before_cursor)
```

Comment:
If we decide to go with SqlStatement() as the classname this should also be renamed to `stmt` instead of `doc`.